app/rankers/algorithms.py

```diff
@@ -113,12 +113,57 @@ def thompson_sampling(
 
 
 def personalize_topic_slates(input_slate_configs: List['SlateConfigModel'],
+                             personalized_topics: PersonalizedTopicList,
+                             topic_limit: Optional[int] = 1) -> List['SlateConfigModel']:
+    """
+    This routine takes a list of slates as input in which must include slates with an associated curator topic
+    label.  It uses the topic_profile that is supplied by RecIt to re-rank the slates according to affinity
+    with items in the user's list.
+    This version allows non-topic slates within the lineup.  These are left in order in the output configs
+    list.  Personalizable (topic) slates are re-ordered using their initial slots in the config lineup.
+    If the topic_limit parameter is included this will determine the number of topic slates that
+    remain in the output config list.
+    :param input_slate_configs: SlateConfigModel list that includes slates with curatorTopicLabels
+    :param personalized_topics: response from RecIt listing topics ordered by affinity to user
+    :param topic_limit: desired number of topics to return, if this is set the number of slates returned is truncated.
+                        otherwise all personalized topics among the input slate configs are returned
+    :return: SlateLineupExperimentModel with reordered slates
+    """
+    topic_to_score_map = {t.curator_topic_label: t.score for t in personalized_topics.curator_topics}
+    # filter non-topic slates
+    personalizable_configs = list(filter(lambda s: s.curator_topic_label in topic_to_score_map, input_slate_configs))
+    print(personalizable_configs)
+    if not personalizable_configs:
+        raise ValueError(f"Input lineup to personalize_topic_slates includes no topic slates")
+    elif topic_limit and len(personalizable_configs) < topic_limit:
+        raise ValueError(f"Input lineup to personalize_topic_slates includes fewer topic slates than requested")
+    # re-rank topic slates
+    personalizable_configs.sort(key=lambda s: topic_to_score_map.get(s.curator_topic_label), reverse=True)
+
+    output_configs = list()
+    added_topic_slates = 0
+    personalized_index = 0
+    for config in input_slate_configs:
+        if config in personalizable_configs:
+            # if slate is personalizable add highest ranked slate remaining
+            if added_topic_slates < topic_limit:
+                output_configs.append(personalizable_configs[personalized_index])
+                added_topic_slates += 1
+                personalized_index += 1
```

Comment:
I've come to realize I'm baised towards Pythonic solutions to a fault.

I was thinking you could probably do this in a few lines, but taking the indices for the personalizable configs, sorting them based on the score, and then sorting everything based on the ranked indices. However, a solution using a for-loop is probably more readable in this case.